compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/expression/FirWhenConditionChecker.kt

```diff
@@ -5,21 +5,37 @@
 
 package org.jetbrains.kotlin.fir.analysis.checkers.expression
 
+import org.jetbrains.kotlin.KtNodeTypes
 import org.jetbrains.kotlin.fir.analysis.checkers.checkCondition
 import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
 import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
 import org.jetbrains.kotlin.fir.analysis.diagnostics.withSuppressedDiagnostics
+import org.jetbrains.kotlin.fir.analysis.getChild
+import org.jetbrains.kotlin.fir.expressions.FirWhenBranch
 import org.jetbrains.kotlin.fir.expressions.FirWhenExpression
 import org.jetbrains.kotlin.fir.expressions.impl.FirElseIfTrueCondition
+import org.jetbrains.kotlin.lexer.KtTokens
 
 object FirWhenConditionChecker : FirWhenExpressionChecker() {
     override fun check(expression: FirWhenExpression, context: CheckerContext, reporter: DiagnosticReporter) {
+        val hasSubject = expression.subject != null
         for (branch in expression.branches) {
             val condition = branch.condition
             if (condition is FirElseIfTrueCondition) continue
             withSuppressedDiagnostics(condition, context) {
                 checkCondition(condition, context, reporter)
             }
+            if (!hasSubject) {
+                checkCommaInBranchCondition(branch, context, reporter)
+            }
+        }
+    }
+
+    private fun checkCommaInBranchCondition(branch: FirWhenBranch, context: CheckerContext, reporter: DiagnosticReporter) {
+        if (branch.source?.elementType == KtNodeTypes.WHEN_ENTRY && branch.source?.getChild(KtTokens.COMMA, depth = 1) != null) {
+            reporter.reportOn(branch.source, FirErrors.COMMA_IN_WHEN_CONDITION_WITHOUT_ARGUMENT, context)
```

Comment:
Common expression can be extracted:

```suggestion
        val source = branch.source
        if (source != null && source.elementType == KtNodeTypes.WHEN_ENTRY && source.getChild(KtTokens.COMMA, depth = 1) != null) {
            reporter.reportOn(source, FirErrors.COMMA_IN_WHEN_CONDITION_WITHOUT_ARGUMENT, context)
```